lib/beacon-verification.ts

```diff
@@ -20,43 +20,52 @@ type PointG1 = typeof bls.G1.ProjectivePoint.ZERO
 type PointG2 = typeof bls.G2.ProjectivePoint.ZERO
 
 async function verifyBeacon(chainInfo: ChainInfo, beacon: RandomnessBeacon, expectedRound: number): Promise<boolean> {
-    const publicKey = chainInfo.public_key
-
-    if (beacon.round !== expectedRound) {
-        console.error('round was not the expected round')
-        return false
-    }
-
-    if (!await randomnessIsValid(beacon)) {
-        console.error('randomness did not match the signature')
-        return false
-    }
-
-    if (isChainedBeacon(beacon, chainInfo)) {
-        return bls.verify(beacon.signature, await chainedBeaconMessage(beacon), publicKey)
-    }
-
-    if (isUnchainedBeacon(beacon, chainInfo)) {
-        return bls.verify(beacon.signature, await unchainedBeaconMessage(beacon), publicKey)
-    }
-
-    if (isG1G2SwappedBeacon(beacon, chainInfo)) {
-        return verifySigOnG1(beacon.signature, await unchainedBeaconMessage(beacon), publicKey)
+    try {
+        if (!chainInfo || !chainInfo.public_key) {
+            throw new Error('Invalid chain info: Missing public key');
+        }
+
+        const publicKey = chainInfo.public_key;
+
+        if (!beacon || typeof beacon.round !== 'number') {
+            throw new Error('Invalid beacon: Missing or invalid round number');
+        }
+
+        if (beacon.round !== expectedRound) {
+            throw new Error(`Round mismatch: Expected ${expectedRound}, got ${beacon.round}`);
+        }
+
+        if (!await randomnessIsValid(beacon)) {
+            throw new Error('Invalid beacon: Randomness does not match signature');
+        }
+
+        if (isChainedBeacon(beacon, chainInfo)) {
+            return bls.verify(beacon.signature, await chainedBeaconMessage(beacon), publicKey);
+        }
+
+        if (isUnchainedBeacon(beacon, chainInfo)) {
+            return bls.verify(beacon.signature, await unchainedBeaconMessage(beacon), publicKey);
+        }
+
+        if (isG1G2SwappedBeacon(beacon, chainInfo)) {
+            return verifySigOnG1(beacon.signature, await unchainedBeaconMessage(beacon), publicKey);
+        }
+
+        if (isG1Rfc9380(beacon, chainInfo)) {
+            return verifySigOnG1(beacon.signature, await unchainedBeaconMessage(beacon), publicKey, 'BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_');
+        }
+
+        if (isBn254OnG1(beacon, chainInfo)) {
+            return bn254.verifyShortSignature(beacon.signature, await unchainedBeaconMessage(beacon, keccak_256), publicKey, {
+                DST: 'BLS_SIG_BN254G1_XMD:KECCAK-256_SVDW_RO_NUL_'
+            });
+        }
+
+        throw new Error(`Unsupported beacon type: ${chainInfo.schemeID}`);
+    } catch (error) {
```

Comment:
catching locally thrown errors feels like a bit of a code smell. Also this could be a breaking change for users expecting certain cryptographic errors to throw exceptions